docs/source/guides/build.rst

```diff
@@ -0,0 +1,218 @@
+
+Building Iroha
+==============
+
+In this guide we will learn how to install all dependencies, required to build 
+Iroha and how to build it.
+
+Preparing the environment
+-------------------------
+
+In order to successfully build Iroha, we need to configure the environment. 
+There are several ways to do it and we will describe all of them.
+
+Currently we support Unix-like systems (we are basically targeting popular 
+Linux distros and macOS). If you happen to have Windows or you don't want to 
+spend time installing all dependencies, you might want to consider using Docker
+environment.
+
+.. hint:: Having troubles? Check FAQ section or communicate to us directly, in
+  case you were stuck on something. We don't expect this to happen, but some
+  issues with an environment are possible.
+
+Docker
+^^^^^^
+First of all, you need to install ``docker`` and ``docker-compose``. You can 
+read how to install it on a 
+`Docker's website <https://www.docker.com/community-edition/>`_
+
+.. note:: Please, use the latest available docker daemon and docker-compose. 
+  If you have troubles launching the Docker container — it is likely that those 
+  problems are caused by an outdated version.
+ 
+After it clone the `Iroha repository <https://github.com/hyperledger/iroha>`_ 
+to the directory of your choice.
+
+.. code-block:: shell
+
+  git clone -b develop https://github.com/hyperledger/iroha
+
+After it, you need to run the development environment. Run the script 
+``run-iroha-dev.sh``, contained in the folder ``scripts``: 
+
+.. code-block:: shell
+
+  sh ./iroha/scripts/run-iroha-dev.sh
+
+.. hint:: Please make sure that Docker is running before executing the script
+
+After you execute this script, following things happen:
+
+1. The script checks if you don't have containers with Iroha already running.
+It ends up with reattaching you to interactive shell upon successful completion.
+2. The script will download ``iroha-docker-develop`` and ``postgres`` images. 
+``iroha-docker-develop`` image contains all development dependencies, and is 
+based on top of ``ubuntu:16.04``. ``postgres`` image is required for starting 
+and running Iroha.
+3. Two containers are created and launched.
+4. The user is attached to the interactive environment for development and 
+testing with ``iroha`` folder mounted from the host machine. Iroha folder 
+is mounted to ``/opt/iroha`` in Docker container.
+
+Now your are ready to build Iroha! Please go to `Building Iroha` section.
+
+.. note::  Docker environment will be removed when you detach from the
```

Comment:
Please change this statement. Since PR #1115 has been merged, docker environment would not be removed after detaching from containers' bash. The environment will be removed in case of launch stop-iroha-dev.sh or manual kill and remove of created containers and the network.